packages/@o3r/schematics/src/rule-factories/ng-add/index.ts

```diff
@@ -1,33 +1,72 @@
-import { Rule, SchematicContext, Tree } from '@angular-devkit/schematics';
-import { NgAddPackageOptions, NodePackageNgAddTask } from '../../tasks/index';
+import {chain, externalSchematic, Rule, RuleFactory, SchematicContext, Tree} from '@angular-devkit/schematics';
+import {NodePackageInstallTask} from '@angular-devkit/schematics/tasks';
+import {lastValueFrom} from 'rxjs';
+import {PackageJson} from 'type-fest';
+import {NgAddPackageOptions} from '../../tasks/index';
 
 /**
  * Install via `ng add` a list of npm packages.
  *
  * @param packages List of packages to be installed via `ng add`
  * @param options install options
  */
-export function ngAddPackages(packages: string[], options?: NgAddPackageOptions): Rule {
+export function ngAddPackages(packages: string[], options?: NgAddPackageOptions, packageJsonPath = '/package.json'): Rule {
   const getInstalledVersion = async (packageName: string) => {
     try {
       return (await import(`${packageName}/package.json`)).version;
     } catch (e) {
       return;
     }
   };
-  return async (tree: Tree, context: SchematicContext) => {
-    if (packages.length > 0) {
-      context.logger.info(`'${options?.parentPackageInfo || ''}' - 'ng add' has been launched for the following packages:`);
-      for (const packageName of packages) {
-        const installedVersion = await getInstalledVersion(packageName);
-        if (!installedVersion || options?.version !== installedVersion) {
-          context.logger.info(`Running ng add for: ${packageName}${options?.version ? ' with version: ' + options.version : ''}`);
-          context.addTask(new NodePackageNgAddTask(packageName, options));
+  const getOptions = (packageName: string, context: SchematicContext) => {
+    try {
+      const collection = context.engine.createCollection(packageName);
+      const schemaOptions = (collection.createSchematic('ng-add').description as any).schemaJson?.properties || {};
+      return Object.entries(options || {}).reduce((accOptions, [key, value]: [string, any]) => {
+        if (schemaOptions[key]) {
+          accOptions[key] = value;
+        }
+        return accOptions;
+      }, {});
+    } catch {
+      context.logger.info(`Could not find ${packageName}:ng-add properties`);
+      return {};
+    }
+  };
+  const ruleFactory: RuleFactory<{ packageName: string; version?: string }> = ({packageName, version}) => {
+    return async (tree: Tree, context: SchematicContext) => {
+      try {
+        const installedVersion: string = await getInstalledVersion(packageName);
+        context.logger.info(`installed version of ${packageName}: ${installedVersion || 'undefined'} | expected: ${options?.version || 'latest'} `);
+        if (!installedVersion || version !== installedVersion) {
+          context.logger.info(`Running ng add for: ${packageName}@${options?.version ? ' with version: ' + options.version : ''}`);
```

Comment:
I think there's an extra `@` after packageName.
```suggestion
          context.logger.info(`Running ng add for: ${packageName}${options?.version && ' with version: ' + options.version}`);
```

Reply:
I don't want to have a Running ng add for @o3r/core with version undefined